plugins/flytekit-inference/flytekitplugins/inference/ollama/serve.py

```diff
@@ -0,0 +1,177 @@
+"""
+
+        ollama_service_ready = f"""
+# Wait for Ollama service to be ready
+max_retries = 30
+retry_interval = 1
+for _ in range(max_retries):
+    try:
+        response = requests.get('{self.base_url}')
+        if response.status_code == 200:
+            print('Ollama service is ready')
+            break
+    except requests.RequestException:
+        pass
+    time.sleep(retry_interval)
+else:
+    print('Ollama service did not become ready in time')
+    exit(1)
+"""
+        if self._model_modelfile:
+            encoded_modelfile = base64.b64encode(self._model_modelfile.encode("utf-8")).decode("utf-8")
+
+            if "{inputs" in self._model_modelfile:
+                python_code = f"""
+{base_code}
+import json
+
+with open('/shared/inputs.json', 'r') as f:
+    inputs = json.load(f)
+
+class AttrDict(dict):
+    def __init__(self, *args, **kwargs):
+        super(AttrDict, self).__init__(*args, **kwargs)
+        self.__dict__ = self
+
+inputs = {{'inputs': AttrDict(inputs)}}
+
+encoded_model_file = '{encoded_modelfile}'
+
+modelfile = base64.b64decode(encoded_model_file).decode('utf-8').format(**inputs)
+modelfile = modelfile.replace('{{', '{{{{').replace('}}', '}}}}')
+
+with open('Modelfile', 'w') as f:
+    f.write(modelfile)
+
+{ollama_service_ready}
+
+for chunk in ollama.create(model='{self._model_name}', path='Modelfile', stream=True):
+    print(chunk)
+"""
+            else:
+                python_code = f"""
+{base_code}
+
+encoded_model_file = '{encoded_modelfile}'
+
+modelfile = base64.b64decode(encoded_model_file).decode('utf-8')
+
+with open('Modelfile', 'w') as f:
+    f.write(modelfile)
+
+{ollama_service_ready}
+
+for chunk in ollama.create(model='{self._model_name}', path='Modelfile', stream=True):
+    print(chunk)
+"""
+        else:
+            python_code = f"""
+{base_code}
+
+{ollama_service_ready}
+
+for chunk in ollama.pull('{self._model_name}', stream=True):
+    print(chunk)
+"""
+
+        command = f'python3 -c "{python_code}"'
+
+        self.pod_template.pod_spec.init_containers.append(
+            V1Container(
+                name=container_name,
+                image="python:3.11-slim",
+                command=["/bin/sh", "-c"],
+                args=[f"pip install requests && pip install ollama && {command}"],
+                resources=V1ResourceRequirements(
+                    requests={
+                        "cpu": self._model_cpu,
+                        "memory": self._model_mem,
+                    },
+                    limits={
+                        "cpu": self._model_cpu,
+                        "memory": self._model_mem,
+                    },
+                ),
+                security_context=V1SecurityContext(
+                    run_as_user=0,
+                ),
+                volume_mounts=[
+                    V1VolumeMount(name="shared-data", mount_path="/shared"),
+                    V1VolumeMount(name="tmp", mount_path="/tmp"),
```

Comment:
`/tmp` is the default directory where the input flytefiles are downloaded. i’ve also added logic to always download files to `/tmp` since the volume mount must be specified at compile time.

Reply:
```
            downloaded_file = FlyteFile.from_source(literal.scalar.blob.uri).download()

            tmp_destination = None
            if not downloaded_file.startswith('/tmp'):
                tmp_destination = '/tmp' + os.path.basename(downloaded_file)
                shutil.copy(downloaded_file, tmp_destination)

            inputs[var_name] = tmp_destination or downloaded_file
```

Reply:
i can also use `/shared` but want to avoid the overhead of copying the file every single time.